java/tsfile/src/main/java/org/apache/tsfile/encrypt/EncryptUtils.java

```diff
@@ -109,37 +158,61 @@ public static IEncryptor getDefaultEncryptor() {
         md.update("IoTDB is the best".getBytes());
         md.update(TSFileDescriptor.getInstance().getConfig().getEncryptKey().getBytes());
         dataEncryptKey = md.digest();
-      } catch (Exception e1) {
-        throw new EncryptException("md5 function not found while using md5 to generate data key");
+      } catch (Exception e) {
+        throw new EncryptException(
+            "md5 function not found while using md5 to generate data key", e);
       }
     } else {
-      encryptType = EncryptionType.UNENCRYPTED;
+      encryptType = "org.apache.tsfile.encrypt.UNENCRYPTED";
       dataEncryptKey = null;
     }
-    return IEncryptor.getEncryptor(encryptType, dataEncryptKey);
+    try {
+      Class<?> encryptTypeClass = Class.forName(encryptType);
+      java.lang.reflect.Constructor<?> constructor =
+          encryptTypeClass.getDeclaredConstructor(byte[].class);
+      return ((IEncrypt) constructor.newInstance(dataEncryptKey));
```

Comment:
The constructor may be stored as a field to reduce the number of reflection calls.

Reply:
Actually the constructor here will only be used once.